libraries/standard/mqtt/src/mqtt.c

```diff
@@ -151,6 +169,61 @@ static MQTTStatus_t validateSubscribeUnsubscribeParams( const MQTTContext_t * co
 
 /*-----------------------------------------------------------*/
 
+static MQTTStatus_t receivePacket( MQTTContext_t * const pContext,
+                                   MQTTPacketInfo_t incomingPacket,
+                                   uint32_t remainingTimeMs )
+{
+    return MQTTSuccess;
+}
+
+/*-----------------------------------------------------------*/
+
+static MQTTStatus_t receiveConnack( MQTTContext_t * const pContext,
+                                    MQTTPacketInfo_t * const incomingPacket )
+{
+    MQTTStatus_t status = MQTTSuccess;
+    uint8_t retryCount = 0U;
+
+    assert( pContext != NULL );
+    assert( incomingPacket != NULL );
+
+    do
+    {
+        /* Transport read for incoming CONNACK packet type and length. */
+        status = MQTT_GetIncomingPacketTypeAndLength( pContext->transportInterface.recv,
+                                                      pContext->transportInterface.networkContext,
+                                                      incomingPacket );
+
+        if( status == MQTTSuccess )
+        {
+            /* Reading the remainder of the packet by transport recv.
+             * The timeout is passed as 0 as it is only 2 more bytes to
+             * read for CONNACK. */
+            status = receivePacket( pContext,
+                                    *incomingPacket,
+                                    0U );
```

Comment:
yes. That's correct. Since we have successfully read packet type and remaining length from the network. It will be 2 more bytes to read from the network buffer.

Reply:
It makes sense to use recvExact. However, the check pContext->networkBuffer.size >= 2 should be performed first as recvExact assumes the buffer is large enough for the packet

Reply:
I think it still makes sense to use recvPacket as it has a wrapper logic to verify the length of the buffer and logic to dump the packet.